ydb/library/yql/core/yql_cost_function.cpp

```diff
@@ -65,25 +73,56 @@ TOptimizerStatistics NYql::ComputeJoinStats(const TOptimizerStatistics& leftStat
 
 
     if (IsPKJoin(rightStats,rightJoinKeys)) {
-        newCard = leftStats.Nrows * rightStats.Selectivity;
-        selectivity = leftStats.Selectivity * rightStats.Selectivity;
-        leftKeyColumns = true;
-        if (leftStats.Type == EStatisticsType::BaseTable){
-            outputType = EStatisticsType::FilteredFactTable;
-        } else {
-            outputType = leftStats.Type;
+        bool outputTypeDefined = false;
```

Comment:
This is a bit strange, how does this logic work?
What does outputTypeDefined mean?

This either needs comments or more clear logic. Let's start with comments